subprojects/gradle/impact/src/main/kotlin/com/avito/impact/plugin/ImpactMetricsSender.kt

```diff
@@ -0,0 +1,48 @@
+package com.avito.impact.plugin
+
+import com.avito.android.sentry.Environment
+import com.avito.android.sentry.EnvironmentInfo
+import com.avito.android.stats.GaugeMetric
+import com.avito.android.stats.StatsDSender
+import com.avito.android.stats.graphiteSeries
+import com.avito.impact.ConfigurationType
+import com.avito.impact.ModifiedProjectsFinder
+import com.avito.impact.configuration.internalModule
+
+class ImpactMetricsSender(
+    private val statsDSender: StatsDSender,
+    private val environmentInfo: EnvironmentInfo
+) {
+
+    init {
+        require(environmentInfo.environment is Environment.CI) {
+            "ImpactMetricsSender should run only in CI environment"
+        }
+    }
+
+    private val prefix by lazy {
+        val envName = environmentInfo.environment.publicName
+        val node = environmentInfo.node?.take(32) ?: "_"
+        val buildId = requireNotNull(environmentInfo.teamcityBuildId()) {
+            "ImpactMetricsSender should run only if teamcityBuildInfo available"
+        }
+        graphiteSeries(envName, node, buildId)
+    }
+
+    fun sendModifiedProjectCounters(modifiedProjectsFinder: ModifiedProjectsFinder) {
```

Comment:
finder looks more like a constructor parameter

Reply:
it's not, because `ModifiedProjectsFinder` contains variable: `project`; so impact metrics could be potentially reused for another parts of project;

However problem here goes to `ModifiedProjectsFinder` design which accepts `project` as constructor variable